### ISSUE TITLE
Add a caching layer?

### ISSUE THREAD

**Issue by iandees** [JasonSanford/tileify-ags-proxy#1]
What do you think about adding a very simple caching layer to this service so that the ESRI server doesn't have to deal with repeated requests for the same image?

My use-case here is that I want to run this on OpenStreetMap US's server and let people add their own layers, cache them, and share the imagery with the whole community.

**Comment by JasonSanford** [comment]
Yeah, this seems valuable. This app could bring a slower ArcGIS Server instance to its knees with multiple users hitting it.

I'm not sure what type of caching implementation would make the most sense. Something like [leveldown](https://github.com/Level/levelup) might be fairly easy to implement. Redis would probably work well too, but I don't think the free tier offered by [RedisToGo](https://addons.heroku.com/redistogo) would go very far (5MB). [Memcachier](https://elements.heroku.com/addons/memcachier#addon-plans) would probably work too, and provides 25MB on a free tier.

If some cache was implemented, it would be awesome to make it optional because as  it stands now this app needs little more than node + express to run, which is great for folks who want to stand up one-off instances for small projects.

**Comment by iandees** [comment]
Yep, optional makes sense.

I was thinking about using Postgres. On Heroku they offer 10K rows for free and don't mention a size limit. Although swapping out postgres for memcache/redis would be pretty easy.

I'll give it a try soon and put together a pull request for comment.

**Comment by tmcgee** [comment]
I would be interested in this as well. What about something like [express-tile-cache](https://www.npmjs.com/package/express-tile-cache) npm module? It has built-in support for caching tiles to the filesystem, [redis](https://www.npmjs.com/package/redis-tile-store) and [s3](https://www.npmjs.com/package/s3-tile-storage).

https://github.com/CGastrell/express-tile-cache
